### ISSUE TITLE
Crash in CertVerifyProc

### ISSUE THREAD

**Issue by iefremov** [brave/brave-browser#14407]
https://share.backtrace.io/api/share/px3iyPMUpWXKEWyx167fG7r1
```
[ 00 ] net::(anonymous namespace)::VerifyWithGivenFlags(net::X509Certificate*, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int, bool, net::CRLSet*, net::CertVerifyResult*, net::(anonymous namespace)::CRLSetResult*)
[ 01 ] net::CertVerifyProcMac::VerifyInternal(net::X509Certificate*, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int, net::CRLSet*, std::__1::vector<scoped_refptr<net::X509Certificate>, std::__1::allocator<scoped_refptr<net::X509Certificate> > > const&, net::CertVerifyResult*, net::NetLogWithSource const&)
[ 02 ] net::CertVerifyProc::Verify(net::X509Certificate*, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int, net::CRLSet*, std::__1::vector<scoped_refptr<net::X509Certificate>, std::__1::allocator<scoped_refptr<net::X509Certificate> > > const&, net::CertVerifyResult*, net::NetLogWithSource const&)
[ 03 ] net::(anonymous namespace)::DoVerifyOnWorkerThread(scoped_refptr<net::CertVerifyProc> const&, scoped_refptr<net::X509Certificate> const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int, scoped_refptr<net::CRLSet> const&, std::__1::vector<scoped_refptr<net::X509Certificate>, std::__1::allocator<scoped_refptr<net::X509Certificate> > > const&, net::NetLogWithSource const&)
[ 04 ] base::internal::Invoker<base::internal::BindState<std::__1::unique_ptr<net::(anonymous namespace)::ResultHelper, std::__1::default_delete<net::(anonymous namespace)::ResultHelper> > (*)(scoped_refptr<net::CertVerifyProc> const&, scoped_refptr<net::X509Certificate> const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int, scoped_refptr<net::CRLSet> const&, std::__1::vector<scoped_refptr<net::X509Certificate>, std::__1::allocator<scoped_refptr<net::X509Certificate> > > const&, net::NetLogWithSource const&), scoped_refptr<net::CertVerifyProc>, scoped_refptr<net::X509Certificate>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, int, scoped_refptr<net::CRLSet>, std::__1::vector<scoped_refptr<net::X509Certificate>, std::__1::allocator<scoped_refptr<net::X509Certificate> > >, net::NetLogWithSource>, std::__1::unique_ptr<net::(anonymous namespace)::ResultHelper, std::__1::default_delete<net::(anonymous namespace)::ResultHelper> > ()>::RunOnce(base::internal::BindStateBase*)
[ 05 ] void base::internal::ReturnAsParamAdapter<std::__1::unique_ptr<net::(anonymous namespace)::ResultHelper, std::__1::default_delete<net::(anonymous namespace)::ResultHelper> > >(base::OnceCallback<std::__1::unique_ptr<net::(anonymous namespace)::ResultHelper, std::__1::default_delete<net::(anonymous namespace)::ResultHelper> > ()>, std::__1::unique_ptr<std::__1::unique_ptr<net::(anonymous namespace)::ResultHelper, std::__1::default_delete<net::(anonymous namespace)::ResultHelper> >, std::__1::default_delete<std::__1::unique_ptr<net::(anonymous namespace)::ResultHelper, std::__1::default_delete<net::(anonymous namespace)::ResultHelper> > > >*)
[ 06 ] base::internal::Invoker<base::internal::BindState<void (*)(base::OnceCallback<std::__1::unique_ptr<net::(anonymous namespace)::ResultHelper, std::__1::default_delete<net::(anonymous namespace)::ResultHelper> > ()>, std::__1::unique_ptr<std::__1::unique_ptr<net::(anonymous namespace)::ResultHelper, std::__1::default_delete<net::(anonymous namespace)::ResultHelper> >, std::__1::default_delete<std::__1::unique_ptr<net::(anonymous namespace)::ResultHelper, std::__1::default_delete<net::(anonymous namespace)::ResultHelper> > > >*), base::OnceCallback<std::__1::unique_ptr<net::(anonymous namespace)::ResultHelper, std::__1::default_delete<net::(anonymous namespace)::ResultHelper> > ()>, std::__1::unique_ptr<std::__1::unique_ptr<net::(anonymous namespace)::ResultHelper, std::__1::default_delete<net::(anonymous namespace)::ResultHelper> >, std::__1::default_delete<std::__1::unique_ptr<net::(anonymous namespace)::ResultHelper, std::__1::default_delete<net::(anonymous namespace)::ResultHelper> > > >*>, void ()>::RunOnce(base::internal::BindStateBase*)
[ 07 ] base::(anonymous namespace)::PostTaskAndReplyRelay::RunTaskAndPostReply(base::(anonymous namespace)::PostTaskAndReplyRelay)
[ 08 ] base::internal::Invoker<base::internal::BindState<void (*)(base::(anonymous namespace)::PostTaskAndReplyRelay), base::(anonymous namespace)::PostTaskAndReplyRelay>, void ()>::RunOnce(base::internal::BindStateBase*)

```

**Comment by iefremov** [comment]
This is top1, but I have no idea what's happening. This is only MacOS and happens on shutdown.
cc @rebron @bsclifton

**Comment by iefremov** [comment]
upd: this is a cyclic crash during startup

**Comment by dekay7** [comment]
THIS IS EXACTLY WHAT I WAS LOOKING FOR!!! Sorry I could not originally find it. Thank you so much for guiding me here @iefremov. It is appreciated.

**Comment by iefremov** [comment]
@dekay7 are you using big sure beta?

**Comment by dekay7** [comment]
Yes, I am.

**Comment by iefremov** [comment]
@dekay7 it appears to me the crash is (1) originated in the upstream code (2) is specific to exactly this MacOS version

The current pure guess is that this [1] can fix the problem once it reaches our codebase

[1] https://source.chromium.org/chromium/chromium/src/+/6db1b2f9c4b6b0aaf3f08755badd644e6c78fafa

**Comment by dekay7** [comment]
@iefremov Ok great! Thank you for the information. It's so interesting to me how people can find such small and precise errors within software code.

**Comment by iefremov** [comment]
Should be fixed in 1.22, @dekay7 can you please verify?

**Comment by dekay7** [comment]
I have not had the issue for a while so yes it probably is fixed. Thank you!
